src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -114,14 +113,20 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
             var tagHelpersFromCompilation = declCompilation
                 .Combine(razorSourceGeneratorOptions)
+                .Combine(isGeneratorSuppressed)
                 .Select(static (pair, _) =>
                 {
-                    RazorSourceGeneratorEventSource.Log.DiscoverTagHelpersFromCompilationStart();
 
-                    var (compilation, razorSourceGeneratorOptions) = pair;
+                    var ((compilation, razorSourceGeneratorOptions), isGeneratorSuppressed) = pair;
+                    var results = new List<TagHelperDescriptor>();
```

Comment:
We should switch to ImmutableArray /Builder here at some point too, but it gets wired quite deep into the tag helper discovery machinery, so I'm leaving as-is for now.

Reply:
If only @DustinCampbell had a PR out for that now... 😉.